### PR TITLE
Eliminate reader tests

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Remove ngraph reader test
 - Extend IR frontend and Core tests to cover gaps

### Tickets:
 - CVS-73617

### CODE REVIEW

**Comment by olpipi** [comment]
@ilyachur  please start review. and clarify about these 2 files:
src/tests/functional/inference_engine/ngraph_reader/linear_ops_tests.cpp
src/tests/functional/inference_engine/ngraph_reader/reduce_to_pooling_tests.cpp

There are small fixes for coverage improvement to be uploaded today.

**Comment by ilyachur** [comment]
> @ilyachur please start review. and clarify about these 2 files: src/tests/functional/inference_engine/ngraph_reader/linear_ops_tests.cpp src/tests/functional/inference_engine/ngraph_reader/reduce_to_pooling_tests.cpp
> 
> There are small fixes for coverage improvement and codestyle to be uploaded today.

@itikhono Could you please advise? Should we move these tests to transformations? Or can  we remove these files?

**Comment by itikhono** [comment]
@olpipi @ilyachur as I can see, we already have such tests for transformations
MulAddToScaleShift or Power: openvino/src/tests/functional/inference_engine/transformations/legacy/mul_add_conversion_test.cpp
openvino/src/common/legacy/src/transformations/convert_opset1_to_legacy/convert_mul_add_to_scaleshift_or_power.cpp

ReduceToPooling: openvino/src/tests/functional/inference_engine/transformations/op_conversions/convert_reduce_to_pooling_test.cpp

it looks like we can delete these ngraph reader_tests

File: src/core/tests/visitors/op/abs.cpp [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "unary_ops.hpp"
+
+using Type = ::testing::Types<UnaryOperatorType<ngraph::op::Abs, ngraph::element::f32>>;
```
Comment (ilyachur) [inline src/core/tests/visitors/op/abs.cpp:7]: Please use new API :) It can be fixed separatelly.

**Review by ilyachur — APPROVED** [review]
Please restore custom constant op tests. It can be done in next PR

**Pull request merged**